src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -323,21 +323,15 @@ function ReportActionItemMessageEdit(props) {
                             onFocus={() => {
                                 setIsFocused(true);
                                 reportScrollManager.scrollToIndex({animated: true, index: props.index}, true);
-                                ComposerActions.setShouldShowComposeInput(false);
+                                setShouldShowComposeInputKeyboardAware(false);
                             }}
                             onBlur={(event) => {
                                 setIsFocused(false);
                                 const relatedTargetId = lodashGet(event, 'nativeEvent.relatedTarget.id');
-
-                                // Return to prevent re-render when save/cancel button is pressed which cancels the onPress event by re-rendering
-                                if (_.contains([saveButtonID, cancelButtonID, emojiButtonID], relatedTargetId)) {
```

Comment:
This caused a regression. We should revert the changes related to `emojiButtonID` because that button will open a modal and eventually the focus will be shifted to that modal.

- Previously: input is focused -> button is clicked -> input loses focus -> **skip showing main composer because the focus is lost to the emojibutton - detected through nativeID** -> button is focused -> open modal -> modal is focused. If we close the modal, the main composer wouldn't be visible.

- Now: input is focused -> button is clicked -> open modal -> input loses focus -> **show main composer because the focus is lost to something that we don't put on our "blacklist"** -> modal is focused. If we close the modal the main composer would be visible but focus will be shifted to the edit input and cause a flicker effect.

Reply:
@s77rt 
I think we need to remove `onMouseDown` as well from `EmojiPickerButton` because preventing mouse down default makes the `relatedTarget.id = undefined`

Reply:
Yes, we do need to remove it. We need either onMouseDown approach or the blacklist native id approach. Sorry if that was not clear in my previous comment.

Reply:
@s77rt 

Reverted. Please check